### ISSUE TITLE
Benchmark training over coco dataset

### ISSUE THREAD

**Issue by eddyxu** [eto-ai/lance#156]
# Problem Statement

Implement a training benchmark for object detection dataset, i.e., `COCO`. 

We can use this benchmark to evaluate the developer experience for MLE to use lance, also, since object detection datasets are usually highly nested, we can test the performance of lance over nested fields (i.e., label / segmentation) as well. 

# Desired Behavior

Use pytorch or Pytorch Ligntning to build a training script, test against: 

* lance / parquet / raw json format
* From Local disk and from S3.
* Data exploration via `duckdb`.
